client/dynamicplugins/registry.go

```diff
@@ -301,6 +303,46 @@ func (d *dynamicRegistry) ListPlugins(ptype string) []*PluginInfo {
 	return plugins
 }
 
+// WaitForPlugin repeatedly checks until a plugin with a given type and name
+// becomes available or its context is canceled or times out.
+// Callers should pass in a context with a sensible timeout
+// for the plugin they're expecting to find.
+func (d *dynamicRegistry) WaitForPlugin(ctx context.Context, ptype, name string) (*PluginInfo, error) {
+	// these numbers are almost arbitrary...
+	delay := 200     // milliseconds between checks, will backoff
+	maxDelay := 5000 // up to 5 seconds between each check
+
+	// put a long upper bound on total time,
+	// just in case callers don't follow directions.
+	ctx, cancel := context.WithTimeout(ctx, 24*time.Hour)
+	defer cancel()
+
+	timer := time.NewTimer(time.Duration(delay) * time.Millisecond)
```

Comment:
There's a `NewSafeTimer` function in `helper/funcs.go` that returns a timer and a stop function that you can defer. That makes it more obvious when we're accidentally leaking the timer's resources by not calling `timer.Stop`.

The other thing that's nice about `NewSafeTimer` is that we can start with a "zero" delay (1ms internally), so that we're not delaying by 200ms for no reason if the plugin is already started (as will be the common case whenever a new allocation is placed).

_That_ being said, because "plugin is already up" is the most typical path, we could probably check `ListPlugins` once before even setting up the context and timer and avoid spinning up the extra goroutine at all!